File: app/src/main/java/app/com/ezata/adapter/HistoryAdapter.java
package app.com.ezata.adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import app.com.ezata.OnItemClicked;
import app.com.ezata.R;
import app.com.ezata.model.HistoryDetails;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {
    HistoryDetails[] historyDetails;
    OnItemClicked onItemClicked;

    public HistoryAdapter(HistoryDetails[] historyDetails, OnItemClicked onItemClicked) {
        this.historyDetails = historyDetails;
        this.onItemClicked = onItemClicked;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem = layoutInflater.inflate(R.layout.history_item, parent, false);
        ViewHolder viewHolder = new HistoryAdapter.ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        HistoryDetails details = historyDetails[position];
        holder.tvDate.setText(details.getDate());
        holder.tvTime.setText(details.getTime());
        holder.tvOrderNmber.setText(details.getOrderNumber());
        holder.tvCustomerName.setText(details.getCustomerName());
        holder.tvType.setText(details.getType());
        holder.tvAmount.setText(details.getAmount());
        holder.tvStatus.setText(details.getStatus());
        holder.fullSize.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onItemClicked.onItemSelect(position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return historyDetails.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvDate, tvTime, tvOrderNmber, tvCustomerName, tvType, tvAmount, tvStatus;
        ConstraintLayout fullSize;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDate = itemView.findViewById(R.id.tvDate);
            tvTime = itemView.findViewById(R.id.tvTime);
            tvOrderNmber = itemView.findViewById(R.id.tvOrderNumber);
            tvCustomerName = itemView.findViewById(R.id.tvCustomerName);
            tvType = itemView.findViewById(R.id.tvType);
            tvAmount = itemView.findViewById(R.id.tvAmount);
            tvStatus = itemView.findViewById(R.id.tvStatus);
            fullSize = itemView.findViewById(R.id.fullsize);
        }
    }
}

File: app/src/main/java/app/com/ezata/adapter/HistoryAdapter.kt
package app.com.ezata.adapter

import android.annotation.SuppressLint
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import app.com.ezata.OnItemClicked
import app.com.ezata.R
import app.com.ezata.model.parse.OrdersParse
import java.text.SimpleDateFormat
import java.util.*
import kotlin.collections.ArrayList

class HistoryAdapter(
    var historyDetails: ArrayList<OrdersParse?>,
    var onItemClicked: OnItemClicked
) :
    RecyclerView.Adapter<HistoryAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem =
            layoutInflater.inflate(R.layout.history_item, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, @SuppressLint("RecyclerView") position: Int) {
        val details = historyDetails[position]
//        holder.tvDate.text = getDate(details?.createdAt.toString())
        if (position == 0) {
            holder.tvDate.text = getDate("2022-07-16 12:10:30")
        }
        holder.tvTime.text = getTime(details?.createdAt.toString())
        holder.tvOrderNmber.text = details?.orderNumber ?: "N/A"
        holder.tvAmount.text = ("$" + String.format("%.2f", details?.amount)) ?: "N/A"
        holder.tvStatus.text = details?.orderStatus.toString() ?: "N/A"

        details?.user?.fetchIfNeeded()?.let {
            holder.tvCustomerName.text =
                "${it.getString("first_name") ?: ""} ${it.getString("last_name") ?: ""}" ?: "N/A"
        }

        holder.fullSize.setOnClickListener { details?.let { it1 -> onItemClicked.onItemSelect(it1) } }
    }


    override fun getItemCount(): Int {
        return historyDetails.size
    }

    private fun getDate(orderDate: String): String {
        /*val dateFormatter = SimpleDateFormat("EEE MMM dd HH:mm:ss zzzz yyyy")
        val date: Date? = dateFormatter.parse(orderDate)

        val formatter2 = SimpleDateFormat("yyyy-MM-dd")

        return formatter2.format(date)*/
        val dateFormatter = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
        val dateFormatter2 = SimpleDateFormat("EEE MMM dd HH:mm:ss zzzz yyyy")
        val date: Date? = dateFormatter.parse(orderDate)

        val diff = dateFormatter2.parse(Date().toString())?.time?.minus(date?.time ?: 0)

        val seconds = diff?.div(1000)
        val minutes = seconds?.div(60)
        val hours = minutes?.div(60)
        val days = hours?.div(24)

        Log.d("Diff: ", orderDate)
        Log.d(
            "Diff: ",
            days.toString() + " Days, " + hours + " Hours, " + minutes + " Minutes, " + seconds + " Seconds"
        )
        /*val dateFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
        val date: Date? = dateFormatter.parse(orderDate)

        val formatter2 = SimpleDateFormat("yyyy-MM-dd")

        return formatter2.format(date)*/

        return ""
    }

    private fun getTime(orderDate: String): String {
        val dateFormatter = SimpleDateFormat("EEE MMM dd HH:mm:ss zzzz yyyy")
        val date: Date? = dateFormatter.parse(orderDate)

        val formatter2 = SimpleDateFormat("HH:mm")

        return formatter2.format(date)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvDate: TextView = itemView.findViewById(R.id.tvDate)
        var tvTime: TextView = itemView.findViewById(R.id.tvTime)
        var tvOrderNmber: TextView = itemView.findViewById(R.id.tvOrderNumber)
        var tvCustomerName: TextView = itemView.findViewById(R.id.tvCustomerName)
        var tvType: TextView = itemView.findViewById(R.id.tvType)
        var tvAmount: TextView = itemView.findViewById(R.id.tvAmount)
        var tvStatus: TextView = itemView.findViewById(R.id.tvStatus)
        var fullSize: ConstraintLayout = itemView.findViewById(R.id.fullsize)
    }
}